easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.5.0-foss-2021a-CUDA-11.3.1.eb

```diff
@@ -0,0 +1,240 @@
+easyblock = 'PythonBundle'
+
+name = 'TensorFlow'
+version = '2.5.0'
```

Comment:
We should update to either 2.5.1 or 2.6.0.

Reply:
I'd go directly to 2.6.0, but yes, 2.5.1 is minimum
We could also explore using TF 2.5 modules (i.e. only major and minor version) as TF releases bugfixes for the patch levels (usually security stuff), e.g. there is a new 2.3.4 release. Similar to how we do Java

Reply:
True, I didn't dare touch 2.6.0 yet tbh. I had to bring this to `2021` at our site a.s.a.p (getting a new cluster and don't want to install `2020` if we don't have to), but don't really have the time to dive into all of the issues that come with TF right now :(

I could definitely try `2.5.1`, that shouldn't be too different in the build procedure (I hope). But I have fundamental problems bringing TF to `2021` because of issues between older numpy & newer OpenBLAS, newer numpy & TF (see above). I'm simply not sure how to move forward with this: if we want this in `2021`, it HAS to be a newer numpy (`1.20.X`), or else it won't work with `OpenBLAS-0.3.15`. But TF is not compatible with it, and I haven't been able to create a patch that makes all of the test suite pass... I simply don't know how to get out of this conondrum. So if anyone has ideas there, let me know =)

In the end, for our local site, I might resort to reinstalling `2020` on the new cluster after all, if only for TensorFlow...

Reply:
Looks I need to dive into TF again...
If there really are issues with numpy vs toolchain vs TF I think then this TF shouldn't be in 2021 as simply "incompatible" (yeah, bit pragmatic here)
I'll look into 2.6.0 for 2021a. For now I'd really suggest to go with 2020b. It's not too bad...
Let's still keep this open and I'll come back to it, once I know more about 2.6.0

Reply:
I have created https://github.com/easybuilders/easybuild-easyconfigs/pull/13877 and that seems to work fine so far. It also has an updated Patch for the numpy issue which should also work in TF 2.5 if we still want that. Otherwise I'd just go with the 2.6 for now